core/src/main/scala/org/apache/spark/util/MutableURLClassLoader.scala

```diff
@@ -82,14 +84,9 @@ private[spark] class ChildFirstURLClassLoader(urls: Array[URL], parent: ClassLoa
   }
 
   override def getResources(name: String): Enumeration[URL] = {
-    val urls = super.findResources(name)
-    val res =
-      if (urls != null && urls.hasMoreElements()) {
-        urls
-      } else {
-        parentClassLoader.getResources(name)
-      }
-    res
+    val childUrls = super.findResources(name).asScala
+    val parentUrls = parentClassLoader.getResources(name).asScala
+    (childUrls ++ parentUrls).asJavaEnumeration
```

Comment:
I get this, and it makes sense, but in the case you describe, the child classloader had no resource under this name. It seems like it would have already returned the parent's copy. Is this really the nature of the problem then or do I misunderstand?